### PR TITLE
[USH-3370] Responsive alignment of case list and map in webapps for different devices

### PR DESCRIPTION
**Pull request by akashkj**
## Product Description
Positioning of map along with case search table based on the screen size.

image

## Technical Summary
https://dimagi-dev.atlassian.net/browse/USH-3370

Checks for the screen width in pixels and defines the alignment of case search table and map based on that. For small screens, map is stacked on top of the case search table while for larger screens, they are stacked in the same row with a split of 2:1 for table and map.

A screen width of 992px is used to define the type of screen, based on https://dimagi-dev.atlassian.net/browse/USH-3370?focusedCommentId=291757

## Feature Flag
NA

### Safety story
Tested changes on staging for both mobile and desktop device. Used emulator to test on iOS with safari browser.

### Automated test coverage
NA

### QA Plan
NA

### Migrations
NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/templates/formplayer/case_list.html [hunk 1]
```diff
@@ -34,64 +34,90 @@ <h1  aria-label="<%- title %>" tabindex="0" class="page-title"><%- title %></h1>
...
-                <ul class="dropdown-menu dropdown-menu-right" role="menu" aria-labelledby="case-list-sort-by">
-                  <%  _.each(headers, function(header, index) { %>
-                    <% if (columnSortable(index)) { %>
-                    <li tabindex="0" class="header-clickable formplayer-request" data-id="<%- index %>" role="menuitem"><a><%= header %></a></li>
-                    <% } %>
-                  <% }); %>
-                </ul>
-              </div>
+      <div class="container-md">
+        <% if (window.innerWidth < 992 && showMap) { %>
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/templates/formplayer/case_list.html:38]: There is a constant `SMALL_SCREEN_WIDTH_PX` for that

File: corehq/apps/cloudcare/templates/formplayer/case_list.html [hunk 2]
```diff
@@ -34,64 +34,90 @@ <h1  aria-label="<%- title %>" tabindex="0" class="page-title"><%- title %></h1>
...
                     <% } %>
-                  <% }); %>
-                </tr>
-              </thead>
+                    <tbody class="wrapper js-case-container">
+                    </tbody>
+                  </table>
+                </div>
+                <% } %>
+              </div>
```
Comment (nospame) [inline corehq/apps/cloudcare/templates/formplayer/case_list.html:109]: This looks like it's closing the same `div` as line 121, and seems to make the pagination controls appear outside the container (normally they're visually part of the case list and share its white background):
image

File: corehq/apps/cloudcare/templates/formplayer/case_list.html [hunk 3]
```diff
@@ -34,64 +34,90 @@ <h1  aria-label="<%- title %>" tabindex="0" class="page-title"><%- title %></h1>
...
-          </table>
+        <% if (window.innerWidth >= 992 && showMap) { %>
+          <div class="col-md-4">
+              <div id="module-case-list-map" class="sticky sticky-map <% if (useTiles) { %> white-border<% } %> noprint-sub-container"></div>
+          </div>
         </div>
         <% } %>
+
+
+
```
Comment (nospame) [inline corehq/apps/cloudcare/templates/formplayer/case_list.html:120]: nit: too many blank lines

**Review by nospame — COMMENTED** [review]
I'll start off by saying the root issue this deals with seems to have been resolved in https://github.com/dimagi/commcare-hq/pull/33351 and there was probably some miscommunication about whether or not it was still a problem, so I'm looking at this PR basically as refactoring.

I see what this going for and I like some of the organizational changes; I also find it easier to follow the flow of logic when the template is structured this way. However, since the conditional checks on lines 38, 46, etc. are only performed once, this only determines layout on first load and I don't think is actively responsive to screen size. The Bootstrap classes from #33351 (e.g. `col-md-5 col-md-push-9`) do provide that responsiveness, as ugly as they are to read, so I think they're preferable in this case.

It also seems like some of the "sticky" map functionality disappears with the changes here, likely because of the container structure (`position: sticky` items only "stick" as long as their containing element is still onscreen). Looking on staging, the map only seems visually sticky when using split screen case search, and (to my knowledge) the map should always be sticky on small screens.

Given all that, my vote would be for this PR to be closed without merging, but happy to hear arguments otherwise.

**Comment by akashkj** [comment]
> my vote would be for this PR to be closed without merging

@nospame Thanks for the review. I am ok with closing this PR in favor of existing code that is responsive to screen size change and 'sticky' behavior of maps.

One point worth noting is that with these changes, for large screen, case list is able to show more columns than the what is visible currently.

E.g. with existing code
image

With PR code change
image

Though this is minor UI improvement, it may be kept in mind as a scope of improvement in future.

I'll close this PR.

**Comment by nospame** [comment]
> One point worth noting is that with these changes, for large screen, case list is able to show more columns than the what is visible currently.

That's a good point and one I hadn't noticed. I assume it's a result of using an 8/4 bootstrap column split instead of the 7/5 there is now. Kim could probably advise on whether it's worth updating on its own -- I think it would just need updates to the [bootstrap classes here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/templates/formplayer/case_list.html#L38-L40), changing to something along the lines of `col-md-4 col-md-push-10` and `col-md-8 col-md-pull-4`

**Comment by akashkj** [comment]
https://github.com/dimagi/commcare-hq/pull/33638#pullrequestreview-1700524452

**Pull request closed without merging**